plugins/inputs/smartctl/smartctl.go

```diff
@@ -0,0 +1,447 @@
+// +build linux
+
+// Package smartctl is a collector for S.M.A.R.T data for HDD, SSD + NVMe devices, linux only
+// https://www.smartmontools.org/
+package smartctl
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Disk is the struct to capture the specifics of a given device
+// It will store some basic information plus anything extended queries return
+type Disk struct {
+	Name      string
+	Vendor    string
+	Product   string
+	Block     string
+	Serial    string
+	Rotation  string
+	Transport string
+	Health    string
+	ReadCache string
+	Writeback string
+	RawData   bytes.Buffer
+	Stats     DiskStats
+}
+
+// DiskFail is the struct to return from our goroutine parseDisks
+type DiskFail struct {
+	Name  string
+	Error error
+}
+
+// DiskStats is the set of fields we'll be collecting from smartctl
+type DiskStats struct {
+	CurrentTemp float64
+	MaxTemp     float64
+	ReadError   []float64
+	WriteError  []float64
+	VerifyError []float64
+}
+
+// SmartCtl is the struct that stores our disk paths for checking
+type SmartCtl struct {
+	Init       bool
+	SudoPath   string
+	CtlPath    string
+	Include    []string
+	Exclude    []string
+	Disks      []string
+	DiskOutput map[string]Disk
+	DiskFailed map[string]error
+}
+
+// tagFinders is a global map of Disk struct elements to corresponding regexp
+var tagFinders = map[string]*regexp.Regexp{
+	"Vendor":    regexp.MustCompile(`Vendor:\s+(\w+)`),
+	"Product":   regexp.MustCompile(`Product:\s+(\w+)`),
```

Comment:
In 6.5: `Model Family:     Western Digital Red` and `Device Model:     WDC WD30EFRX-68EUZN0`